### PR TITLE
Makes mobile an optional parameter in /notify/positive endpoint

### PR DESCRIPTION
**Pull request by dennisgove**
If a mobile number is not provided then an SMS will not be sent. If a mobile number is provided and is valid but an error occurs while sending an SMS then the code will still be returned to the caller with an indication that the SMS was not sent.

### CODE REVIEW

**Comment by dennisgove** [comment]
NYS DoH would like to make optional the sending of an SMS message when generating a new verification code. This PR refactors a bit of the code in that endpoint to make that possible. It also prevents SMS failures from blocking the generation of verification codes. The thinking here is that if the SMS fails to send it's still possible to communicate the verification code. To support this the response has been enhanced to indicate whether the SMS was sent.
```
{
  code "verification_code",
  error "description of potential error",
  expiresAt "timestamp of when code expires, eg 2020-08-21T18:48:26.862Z"
  expiresAtTimestamp: Unix timestamp of expiration, eg 1598035706
  smsSent true/false
}
```

This also logs additional metrics:
 `SMS_FAIL`: Unable to parse mobile or unable to add to the SQS queue
 `SMS_SENT`: Added SMS message to SQS queue
 `VERIFICATION_CODE_GENERATED`: generated and returned verification code
 `VERIFICATION_CODE_FAIL`: unhandled error, verification code not returned

**Comment by dennisgove** [comment]
I should mention that the endpoint path `/notify/positive` isn't ideal for what is actually going on (even before this PR). The true purpose of this endpoint is to generate a verification code. IMHO notification should be independent of that. 

However, I've not created a new endpoint in order to stay backward compatible with existing installs of CovidGreen. I'm open to creating a new endpoint if we think that's a better approach.

File: .env.sample [hunk 1]
```diff
@@ -20,3 +20,5 @@ CODE_CHARSET=0123456789
 CODE_LIFETIME_MINS=10
 DEFAULT_COUNTRY_CODE=
 SMS_QUEUE_URL=
+
+CODE_LIFETIME_MINS=60
```
Comment (colmharte) [inline .env.sample]: This is already defined above

File: lib/routes/notify/schema.js [hunk 2]
```diff
@@ -2,26 +2,16 @@ const S = require('fluent-schema')
...
-        .minLength(5)
-        .maxLength(50)
-    )
-    .prop(
-      'phone',
-      S.string()
-        .minLength(5)
-        .maxLength(50)
-    )
+    .prop('mobile', S.anyOf([S.string().pattern(/^\+?[0-9]{7,15}$/), S.null()]))
```
Comment (colmharte) [inline lib/routes/notify/schema.js]: These too rigid, in some envs the number passed through can also contain additional text, needs to be same validation rules as before

**Review by colmharte — CHANGES_REQUESTED** [review]

File: lib/routes/notify/schema.js (hunk 2 above)
Comment (dennisgove) [reply lib/routes/notify/schema.js]: I copied these from [covid-green-backend-api](https://github.com/project-vagabond/covid-green-backend-api/blob/current/lib/routes/callback/schema.js#L9) but I can back this change out.

Comment (colmharte) [reply lib/routes/notify/schema.js]: Yeah the requirement for this api is different, backend api should always receive consistently formatted numbers, this API doesn't

**Review by colmharte — APPROVED** [review]

**Pull request merged**